downloader/src/main/kotlin/vcs/GitRepo.kt

```diff
@@ -32,7 +32,20 @@ import java.io.IOException
 object GitRepo : GitBase() {
     override val aliases = listOf("gitrepo", "git-repo", "repo")
 
-    override fun getWorkingTree(vcsDirectory: File) = super.getWorkingTree(File(vcsDirectory, ".repo/manifests"))
+    override fun getWorkingTree(vcsDirectory: File) =
+            object : GitWorkingTree(File(vcsDirectory, ".repo/manifests")) {
+                // Return the directory in which "repo init" was run (that directory in not managed with Git).
+                override fun getRootPath() = vcsDirectory.absolutePath
+
+                override fun getPathToRoot(path: File): String {
+                    // GitRepo is special in that the path to the root is supposed to return the path to the manifest
+                    // file in use, which is symlinked from ".repo/manifest.xml". So resolve that path to the underlying
+                    // manifest inside the "manifests" directory.
+                    val manifestLink = File(path, ".repo/manifest.xml")
+                    val manifestDir = File(path, ".repo/manifests")
+                    return manifestLink.canonicalFile.toRelativeString(manifestDir)
```

Comment:
I don't understand this implementation, doesn't `File(path, ".repo/manifest.xml")`assume that `path` is a directory which contains ".repo/manifest.xml"? Wouldn't this fail if any other file than the root directory where `repo init` was called is passed to the function?
It would be great to provide some examples in a test, currently GitRepo is the only VCS we don't have tests for.

Reply:
True, there are some assumptions involved, I'll fix that. While I agree we should add tests here, too, that will probably take too much time for me right now.